boilerplate/openshift/custom-catalog-osd-operator/custom-catalog-build-push.sh

```diff
@@ -30,14 +31,37 @@ function build_catalog_image() {
   fi
 
   ${CONTAINER_ENGINE} pull ${bundle_image}
-  ${opm_local_executable} index add --bundles ${bundle_image} --tag ${image_tag} \
-    --container-tool ${CONTAINER_ENGINE_SHORT}
+  # some upstream bundles images are built specifying a "replaces" field, which means it builds on top of the 
+  # previous minor version. we need to check if the build fails for that reason and use another method
+  echo "building catalog image..."
+  BUILD=$(${opm_local_executable} index add \
+    --bundles ${bundle_image} \
+    --tag ${image_tag} \
+    --container-tool ${CONTAINER_ENGINE_SHORT} 2>&1)
+  
+  ERR_COUNT=$(echo $BUILD | grep -c 'non-existent replacement')
+  
+  # Dump output of build since its stderr and won't show with the variable capture in place
+  echo -e "\n$BUILD\n"
 
+  if [[ ${ERR_COUNT} > 0 ]]; then 
+    echo "adding bundle failed -- bundle specifies a non-existent replacement"
+    echo "re-attempting by using the previous catalog image as the index"
+    
+    # grab the latest catalog image tag to use as an arg to --from-index
+    CATALOG_LATEST_TAG=$(skopeo list-tags docker://${BASE_IMAGE_PATH} | jq -r '.Tags[-1]')
+    ${opm_local_executable} index add \
+      --bundles ${bundle_image} \
+      --tag ${image_tag} \
+      --container-tool ${CONTAINER_ENGINE_SHORT} \
+      --from-index "${BASE_IMAGE_PATH}:${CATALOG_LATEST_TAG}"
+  fi
 }
 
-[[ $# -eq 1 ]] || usage
+[[ $# -eq 2 ]] || usage
 
 REGISTRY_IMAGE_URI=$1
+BASE_IMAGE_PATH=$2
```

Comment:
Instead of needing to add a new argument, you *could* parse the tag off of `$REGISTRY_IMAGE_URI` instead:
```suggestion
BASE_IMAGE_PATH=${REGISTRY_IMAGE_URI%:*}
```

Reply:
I was thinking of doing this originally and honestly didnt know which solution was better/cleaner so I just picked one. Fine with this method too, I didn't have a preference.

Reply:
Yeah, your call. Could also make the args base and tag, to avoid the duplication.